### ISSUE TITLE
AppVeyor

### ISSUE THREAD

**Issue by janisozaur** [falltergeist/falltergeist#476]
Now that #475 is merged, you should get yourselves AppVeyor builds.

AppVeyor is the Travis of Windows world. At OpenRCT2 we use it to produce & deploy official builds for inferior platforms.

They already have a selection of some software pre-installed in the images, see https://www.appveyor.com/docs/installed-software

For dependencies you may pursue following solutions:
1. Build them locally and cache (see https://www.appveyor.com/docs/build-cache)
2. Download pre-built versions from elsewhere (at OpenRCT2 we have another repository set up for dependencies which stores pre-compiled binaries for use on Windows, appveyor downloads it for each build, see https://github.com/OpenRCT2/Dependencies especially this part: https://github.com/OpenRCT2/Dependencies/blob/master/build.ps1#L125-L137)
3. Something else.

**Comment by janisozaur** [comment]
Perhaps worth noting: on Windows we produce static builds, so we can simply provide just one binary with all the things baked in.

**Comment by alexeevdv** [comment]
Thanks for the hint. I think that i'll add automated Windows builds in the near future. Doesn't seems difficult

**Comment by 667bdrm** [comment]
Here is conan integration AppVeyor example https://docs.conan.io/en/latest/integrations/appveyor.html

**Comment by 667bdrm** [comment]
Looks like it will be supported soon :) https://github.com/falltergeist/falltergeist/compare/develop...BGforgeNet:appveyor

**Comment by burner1024** [comment]
I got a barebone build working, but the actual falltergeist.exe is crashing on launch, both develop and latest 0.3.1 version. I don't have Visual Studio locally... Need someone to build both on a local machine and clarify whether it's a falltergeist issue or appveyor build specific. No point in submitting a pull until that. Related #550.

**Comment by JanSimek** [comment]
@burner1024 if no one fixes the conan sdl_image dependency (#576) then you could use [vcpkg ](https://github.com/microsoft/vcpkg) instead. It is preinstalled in Appveyor.

**Comment by burner1024** [comment]
@JanSimek I got the impression you were going to pick the build issue up?

**Comment by JanSimek** [comment]
Not at all. You already have the appveyor configuration set up. I just wanted to see if it runs on Windows.

**Comment by burner1024** [comment]
I see. Well, I'll see when I... have time.
Also, I think Travis supports Windows now, and Github Actions are coming, so maybe one of them could be used instead, to avoid extra entities.

**Comment by burner1024** [comment]
I remember now, the latest problem was that [Debug](https://ci.appveyor.com/project/II/falltergeist/build/job/37a35y0hk1vxrsft) configuration builds, while [Release](https://ci.appveyor.com/project/II/falltergeist/builds/28347627/job/g3rkt8g3j9okabl0?fullLog=true) fails with
```
Creating library C:/projects/falltergeist/lib/falltergeist.lib and object C:/projects/falltergeist/lib/falltergeist.exp
TileMap.cpp.obj : error LNK2001: unresolved external symbol "void * __cdecl operator new(unsigned int)" (??2@YAPAXI@Z) [C:\projects\falltergeist\falltergeist.vcxproj]
```
and I'm not really sure what to do about it.

**Comment by JanSimek** [comment]
> APPEND PROPERTY LINK_FLAGS /NODEFAULTLIB:MSVCRT

Update CMakeLists.txt from the develop branch and it should work.

**Comment by burner1024** [comment]
I did merge develop just before building.

**Comment by JanSimek** [comment]
> This branch is 3 commits ahead, 44 commits behind falltergeist:develop.

I don't know what happened, but that appveyor2 branch is not correctly merged. Just compare the CMakeLists.txt from your appveyor2 branch with the one in the develop branch and you'll see the differences.

**Comment by burner1024** [comment]
Yeah, appveyor2 branch was based on master, appveyor branch on develop, and I merged the wrong one.
So, using develop/appveyor, SDL_mixer [wouldn't build](https://ci.appveyor.com/project/II/falltergeist/builds/28375330/job/3wc2kg84rqb97lr9), so I just added a wget for it. Now it fails on [something else](https://ci.appveyor.com/project/II/falltergeist/builds/28375817/job/6cikqv57o86u8icy).
Given that I don't have Visual Studio or Windows, and know exactly zero c++, I think that's about as far as I'll go. Anyone is welcome to pick up.

**Comment by JanSimek** [comment]
Yea, it is kind of confusing that master is not the default branch. I used vcpkg for the dependencies instead of conan and it works well. If they don't fix the sdl_image bug soon I would suggest removing conan altogether because it will only deter potential Windows users when they compile the project and it will crash immediatelly.
